controllers/front/fail.php

```diff
@@ -91,41 +91,30 @@ public function initContent()
     {
         parent::initContent();
 
-        $cartId = Tools::getValue('cartId');
-        $moduleId = Tools::getValue('moduleId');
-        $orderId = Tools::getValue('orderId');
-        $secureKey = Tools::getValue('secureKey');
-
         $orderLink = $this->context->link->getPageLink(
-            'order-confirmation',
+            'order',
             true,
-            null,
-            [
-                'id_cart' => $cartId,
-                'id_module' => $moduleId,
-                'id_order' => $orderId,
-                'key' => $secureKey,
-                'cancel' => 1,
-            ]
+            null
         );
         if (!SaferPayConfig::isVersion17()) {
+            $this->context->cookie->saferpay_payment_canceled_error =
```

Comment:
create same name redirectWithNotifications function and hide conditional logic there